core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -186,15 +186,25 @@ public TransportClientNodesService removeTransportAddress(TransportAddress trans
             if (closed) {
                 throw new IllegalStateException("transport client is closed, can't remove an address");
             }
-            List<DiscoveryNode> builder = new ArrayList<>();
+            List<DiscoveryNode> listNodesBuilder = new ArrayList<>();
             for (DiscoveryNode otherNode : listedNodes) {
                 if (!otherNode.getAddress().equals(transportAddress)) {
-                    builder.add(otherNode);
+                    listNodesBuilder.add(otherNode);
                 } else {
-                    logger.debug("removing address [{}]", otherNode);
+                    logger.debug("removing address [{}] from list nodes", otherNode);
                 }
             }
-            listedNodes = Collections.unmodifiableList(builder);
+            listedNodes = Collections.unmodifiableList(listNodesBuilder);
+            List<DiscoveryNode> nodesBuilder = new ArrayList<>();
+            for (DiscoveryNode otherNode : nodes) {
+                if (!otherNode.getAddress().equals(transportAddress)) {
+                    nodesBuilder.add(otherNode);
+                } else {
+                    logger.debug("removing address [{}] from connected nodes", otherNode);
```

Comment:
I wonder if this should say something more like `disconnecting from node with address [{}]`?